pkg/domain/infra/abi/containers.go

```diff
@@ -0,0 +1,66 @@
+// +build ABISupport
+
+package abi
+
+import (
+	"context"
+
+	"github.com/containers/libpod/libpod/define"
+	"github.com/containers/libpod/pkg/adapter/shortcuts"
+	"github.com/containers/libpod/pkg/domain/entities"
+	"github.com/pkg/errors"
+)
+
+// TODO: Should return *entities.ContainerExistsReport, error
+func (ic *ContainerEngine) ContainerExists(ctx context.Context, nameOrId string) (*entities.BoolReport, error) {
+	_, err := ic.Libpod.LookupContainer(nameOrId)
+	if err != nil && errors.Cause(err) != define.ErrNoSuchCtr {
+		return nil, err
+	}
+	return &entities.BoolReport{Value: err == nil}, nil
+}
+
+func (ic *ContainerEngine) ContainerWait(ctx context.Context, namesOrIds []string, options entities.WaitOptions) ([]entities.WaitReport, error) {
```

Comment:
Should ContainerWait() and ContainerExists() be in separate files?

Reply:
in future, it might be nice to avoid all the rebasing that this could cause ... leaving for now.